ccl/oidcccl: TestOIDCEnabled failed

ccl/oidcccl.TestOIDCEnabled [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/16868508?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/16868508?buildTab=artifacts#/) on release-23.2 @ [a1a1bf97dd897f3f93b08ac8382bca2af2fd04d7](https://github.com/cockroachdb/cockroach/commits/a1a1bf97dd897f3f93b08ac8382bca2af2fd04d7):


```
=== RUN   TestOIDCEnabled
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/3ed8b395931fde05baf66bd2e4d88d97/logTestOIDCEnabled197546267
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:159: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    authentication_oidc_test.go:209: expected to be redirected to root
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/3ed8b395931fde05baf66bd2e4d88d97/logTestOIDCEnabled197546267
--- FAIL: TestOIDCEnabled (13.83s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestOIDCEnabled.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-42150

--- a/pkg/ccl/oidcccl/authentication_oidc_test.go
+++ b/pkg/ccl/oidcccl/authentication_oidc_test.go
@@ -126,22 +126,17 @@ func TestOIDCEnabled(t *testing.T) {
 		NewOIDCManager = realNewManager
 	}()
 
-	// Set minimum settings to successfully enable the OIDC client
 	sqlDB := sqlutils.MakeSQLRunner(db)
 	sqlDB.Exec(t, fmt.Sprintf(`CREATE USER %s with password 'unused'`, usernameUnderTest))
-	sqlDB.Exec(t, `SET CLUSTER SETTING server.oidc_authentication.provider_url = "providerURL"`)
-	sqlDB.Exec(t, `SET CLUSTER SETTING server.oidc_authentication.client_id = "fake_client_id"`)
-	sqlDB.Exec(t, `SET CLUSTER SETTING server.oidc_authentication.client_secret = "fake_client_secret"`)
-	sqlDB.Exec(t, `SET CLUSTER SETTING server.oidc_authentication.redirect_url = "https://cockroachlabs.com/oidc/v1/callback"`)
-	sqlDB.Exec(t, `set cluster setting server.oidc_authentication.claim_json_key = "email"`)
-	sqlDB.Exec(t, `set cluster setting server.oidc_authentication.principal_regex = '^([^@]+)@[^@]+$'`)
-	sqlDB.Exec(t, fmt.Sprintf(`SET CLUSTER SETTING server.http.base_path = "%s"`, basePath))
-	// Setting the `server.oidc_authentication.enabled` cluster setting through
-	// SQL command adds an overhead in updating the `OIDCEnabled` var.
-	// This fails the test under stress as the value of `OIDCEnabled` var
-	// determines the response to the `/oidc/v1/login` API handler.
-	// Hence, it is recommended to override the value directly.
-	// TODO(pritesh-lahoti): Refactor the above OIDC cluster setting statements.
+
+	// Set minimum settings to successfully enable the OIDC client
+	OIDCProviderURL.Override(ctx, &s.ClusterSettings().SV, "providerURL")
+	OIDCClientID.Override(ctx, &s.ClusterSettings().SV, "fake_client_id")
+	OIDCClientSecret.Override(ctx, &s.ClusterSettings().SV, "fake_client_secret")
+	OIDCRedirectURL.Override(ctx, &s.ClusterSettings().SV, "https://cockroachlabs.com/oidc/v1/callback")
+	OIDCClaimJSONKey.Override(ctx, &s.ClusterSettings().SV, "email")
+	OIDCPrincipalRegex.Override(ctx, &s.ClusterSettings().SV, "^([^@]+)@[^@]+$")
+	server.ServerHTTPBasePath.Override(ctx, &s.ClusterSettings().SV, basePath)
 	OIDCEnabled.Override(ctx, &s.ClusterSettings().SV, true)
 
 	testCertsContext := s.NewClientRPCContext(ctx, username.TestUserName())

